### PR TITLE
Update JSON Schema documentation for V2

### PR DESCRIPTION
**Pull request by tpdorsey**
## Related issue number

Closes PYD-116

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @adriangb

### CODE REVIEW

File: docs/usage/json_schema.md
Comment (Kludex) [inline docs/usage/json_schema.md:729]: I think there's an example that we can use from our code source with discriminators that we can give as an example here.

**Comment by tpdorsey** [comment]
please review

File: docs/usage/json_schema.md
Comment (adriangb) [inline docs/usage/json_schema.md:12]: Can we skip documenting this? I think we may want to remove it...

File: docs/usage/json_schema.md
Comment (adriangb) [inline docs/usage/json_schema.md:724]: @dmontagu I defer to you if you want to document this, or document it with a big warning at the top saying it's experimental?

**Review by adriangb — APPROVED** [review]
left some notes but generally looks good!

File: docs/usage/json_schema.md
Comment (dmontagu) [reply docs/usage/json_schema.md:721]: I think let's  hold off on documenting for now but plan to include it with a warning when we add it, maybe shortly. I'd like to give FastAPI a bit of time first.

**Pull request merged**
